gslb/ingestion/gslb_host_rule.go

```diff
@@ -0,0 +1,272 @@
+
+// 	// gslbhr.Status.Status = msg
+// 	// if !gslbutils.PublishGSLBHostRuleStatus {
+// 	// 	return
+// 	// }
+// 	// obj, updateErr := gslbutils.GlobalGslbClient.AmkoV1alpha1().GSLBHostRules(gslbhr.ObjectMeta.Namespace).Update(gslbhr)
+// 	// if updateErr != nil {
+// 	// 	gslbutils.Errf("Error in updating the GSLBHostRule status object %v: %s", obj, updateErr)
+// 	// }
+// }
+
+func isSitePersistenceProfilePresent(profileName string) bool {
+	aviClient := avictrl.SharedAviClients().AviClient[0]
+	uri := "/api/applicationpersistenceprofile?name=" + profileName
+	result, err := aviClient.AviSession.GetCollectionRaw(uri)
+	if err != nil {
+		gslbutils.Errf("Error getting Site Persistent Profiles")
+		return false
+	}
+	if result.Count <= 0 {
+		gslbutils.Errf("Site Persistent Profile " + profileName + " doesnot exist")
+		return false
+	}
+	return true
+}
+
+func isHealthMonitorRefPresent(refName string) bool {
+	aviClient := avictrl.SharedAviClients().AviClient[0]
+	uri := "/api/healthmonitor?name=" + refName
+	result, err := aviClient.AviSession.GetCollectionRaw(uri)
+	if err != nil {
+		gslbutils.Errf("Error getting Health Monitor Refs")
+		return false
+	}
+	if result.Count <= 0 {
+		gslbutils.Errf("Health Monitor " + refName + " doesnot exist")
+		return false
+	}
+	return true
+}
+
+func isThirdPartyMemberSitePresent(siteName string) bool {
+	aviClient := avictrl.SharedAviClients().AviClient[0]
+	uri := "/api/gslb"
+	result, err := aviClient.AviSession.GetCollectionRaw(uri)
+	if err != nil {
+		gslbutils.Errf("Error getting Health Monitor Refs")
+		return false
+	}
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		gslbutils.Errf("Failed to unmarshal GSLB data, err: %v", err)
+	}
+	for _, elem := range elems {
+		gslb := models.Gslb{}
+		err = json.Unmarshal(elem, &gslb)
+		tpms := gslb.ThirdPartySites
+		for _, tpm := range tpms {
+			if *tpm.Name == siteName {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func ValidateGSLBHostRule(gslbhr *gslbhralphav1.GSLBHostRule) error {
+	gslbhrName := gslbhr.ObjectMeta.Name
+	gslbhrSpec := gslbhr.Spec
+	if gslbhrSpec.Fqdn == "" {
+		return errors.New("GSFqdn missing for " + gslbhrName + " GSLBHostRule")
+	}
+	//TTL needs to be an integer
+	if reflect.TypeOf(gslbhrSpec.TTL).Kind() != reflect.Int {
+		return errors.New("value of TTL should an integer for " + gslbhrName + " GSLBHostRule")
+	}
+
+	sitePersistences := gslbhrSpec.SitePersistence
+	for _, site := range sitePersistences {
+		if reflect.TypeOf(site.Enabled).Kind() != reflect.Bool {
+			return errors.New("Site Persistence enable value should be a bool for " + gslbhrName + " GSLBHostRule")
+		}
+		sitePersistenceProfileName := site.ProfileRef
+		if site.Enabled == true && isSitePersistenceProfilePresent(sitePersistenceProfileName) != true {
+			return errors.New("SitePersistence Profile " + sitePersistenceProfileName + " error for " + gslbhrName + " GSLBHostRule")
```

Comment:
Would it be easier to use `return fmt.Errorf(formatted error)` instead of `errors.New`?

Reply:
Replacing it